web/src/HUD.tsx

```diff
@@ -49,6 +49,7 @@ type Resource = {
   RuntimeStatus: string
   ShowBuildStatus: boolean
   TriggerMode: TriggerMode
+  HasPendingChanges: boolean
```

Comment:
is there a reason this has to be a separate boolean rather than just `PendingBuildEdits.len > 0`?

Reply:
Yes, take this scenario:

1. Make a change to a manual resource
2. The resource errors
-- At this point `PendingBuildEdits` doesn't get emptied because a completed built did not go through.

Therefore if we relied only on `PendingBuildEdits` we wouldn't be able to reliably distinguish between "I changed a file" and "a build failed so there's still files that haven't been updated in the cluster".

Does that make sense?

Reply:
ah yes gotcha!

Reply:
...though i might disambiguate the names a little in that case? I assume the difference to the user is that if `PendingBuildEdits.len > 0 && !HasPendingChanges` then we don't offer them a "play" button?

Reply:
Yeah, thing is that `HasPendingChanges` is a term with [specific meaning in the backend](https://github.com/windmilleng/tilt/blob/16349b8ffdb6b152865db699ab902d07e9bd2a57/internal/store/engine_state.go#L387). If we change it here we should change it everywhere, and then that feels like a separate PR.